x/auth/module.go

```diff
@@ -11,55 +11,56 @@ import (
 
 	"github.com/cosmos/cosmos-sdk/client/context"
 	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
 	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
 	"github.com/cosmos/cosmos-sdk/x/auth/client/cli"
 	"github.com/cosmos/cosmos-sdk/x/auth/client/rest"
 	"github.com/cosmos/cosmos-sdk/x/auth/simulation"
-	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
```

Comment:
The idea is that auth doesn't directly rely on vesting. Consider vesting a sub-module.

Reply:
Vesting RegisterInterfaces is needed to be registered in module.go, I couldn't find any alternative as vesting doesn't have a module.go

@aaronc is there a way we can avoid this?

Reply:
The alternative would be to register vesting interfaces in simapp `MakeCodec`. If you look there you'll see that `std.MakeCodec` registers the vesting amino codec. So we could do something similar.

Maybe `std.RegisterInterfaces`? And that would call `sdk.RegisterInterfaces()` too.

Reply:
Not clear on the best direction but auth cannot rely on vesting directly -- we spent a great deal of time trying to separate it out (it's still not close to perfect).

Reply:
I like the idea of using std.RegisterInterfaces, this should help us to registerInterfaces from any other root dirs in future too.

Will make the changes accordingly

Reply:
👍